components/Profile/ProfileCard.tsx

```diff
@@ -0,0 +1,147 @@
+import Button from '../Utils/Button'
+import { profileCard } from '../../styles/profile'
+import bull from '../Utils/Bullet'
+import React, { FC, useState } from 'react'
+// @ts-ignore
+import { UilLocationPoint, UilPen } from '@iconscout/react-unicons'
+import { useRecoilValue } from 'recoil'
+import { userProfileState } from '../../atoms/user'
+import { useMediaQuery } from '@mui/material'
+import UserProfileForm from '../Login/UserProfileForm'
+import Modal from '../Utils/Modal'
+
+interface ProfileCardProps {
+    bio?: string
+    location?: string
+    name?: string
+    profilePic: string
+    username?: string
+    uid?: string
+    joinedAt?: string
+}
+
+export const ProfileCard: FC<ProfileCardProps> = (props) => {
+    const { bio, location, name, profilePic, username, uid, joinedAt } = props
+    // recoil state to check if Profile card is for current user.
+    const { uid: currentUserUid } = useRecoilValue(userProfileState)
+
+    // hook to check is user is no mobile device or not.
+    const isMobile = useMediaQuery('(max-width: 965px)')
+
+    // Modal
+    const [showModal, setShowModal] = useState(false)
+    const openModal = () => {
+        setShowModal(true)
+    }
+    const closeModal = () => {
+        setShowModal(false)
+    }
+
+    return (
+        <>
+            <div className={'flex flex-col items-center'}>
+                <div className={profileCard.mainDiv}>
+                    {/*profile image*/}
+                    <img
+                        className={profileCard.profileImg}
+                        alt={username}
+                        src={profilePic}
+                    />
+                    {/*container for user details*/}
+                    <div className={profileCard.userDetailsDiv}>
+                        {/*User profile name and buttons list*/}
+                        <div className={'flex items-center mb-1'}>
+                            <span className={profileCard.userProfileName}>
+                                {name}
```

Comment:
Do we want to expose this?
People may want to 'hide" behind their user name. We can add a button to toggle it.
Also, should it be name last name?